packages/dm-core-plugins/src/file/FilePlugin.tsx

```diff
@@ -0,0 +1,202 @@
+          address: `$${blobId}`,
+          type: 'dmss://system/SIMOS/Reference',
+        }
+        onUpload(file, reference)
+      })
+      .catch((error: AxiosError<any>) => {
+        console.log(error)
+        const errorResponse =
+          typeof error.response?.data == 'object'
+            ? error.response?.data?.message
+            : error.response?.data
+        const errorMessage = errorResponse || 'Failed to upload file'
+        setError(errorMessage)
+      })
+      .finally(() => setLoading(false))
+  }
+
+  if (error)
+    return (
+      <ErrorGroup>
+        <b>Error</b>
+        <b>
+          Failed uploading file...
+          <div>
+            <code>{error}</code>
+          </div>
+        </b>
+      </ErrorGroup>
+    )
+
+  return (
+    <>
+      <input
+        type="file"
+        ref={textInput}
+        style={{ display: 'none' }}
+        onChange={(event: ChangeEvent<HTMLInputElement>) => handleUpload(event)}
+      />
+      {loading ? (
+        <Button style={{ margin: '0 10px' }}>
+          <Progress.Dots />
+        </Button>
+      ) : (
+        <Button
+          onClick={() => textInput?.current?.click()}
+          style={{ margin: '0 10px' }}
+        >
+          Upload
+        </Button>
+      )}
+      {error && <p style={{ color: 'red' }}>{error}</p>}
+    </>
+  )
+}
+
+const DownloadButton = (props: any) => {
+  const { entity, dataSourceId } = props
+  const [fileUrl, setFileUrl] = useState('')
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState<string | null>(null)
+
+  const dmssAPI = useDMSS()
+
+  useEffect(() => {
+    setError(null)
+    const options: AxiosRequestConfig = {
+      responseType: 'blob',
+    }
+    dmssAPI
+      .blobGetById(
+        {
+          dataSourceId: dataSourceId,
+          blobId: entity.content['address'],
+        },
+        options
+      )
+      .then((response: any) => {
+        const data = response.data
+        const blob = new Blob([data], { type: entity['filetype'] })
+        setFileUrl(window.URL.createObjectURL(blob))
+      })
+      .catch((error: AxiosError<ErrorResponse>) => {
+        console.error(error)
+        setError(error.response?.data.message || 'Failed to fetch file')
+      })
+      .finally(() => setLoading(false))
+  }, [entity])
+
+  if (error)
+    return (
+      <ErrorGroup>
+        <b>Error</b>
+        <b>
+          Failed to load file...
+          <div>
+            <code>{error}</code>
+          </div>
+        </b>
+      </ErrorGroup>
+    )
+
+  if (loading) return <Loading />
+
+  return (
+    <div style={{ padding: '10px' }}>
+      <Button
+        href={fileUrl}
+        rel="noopener noreferrer"
+        download={entity['name']}
+      >
+        Download
+      </Button>
+    </div>
+  )
+}
+
+export const FilePlugin = (props: IUIPlugin) => {
+  const { idReference } = props
+  const [entity, loading, updateDocument, error] = useDocument<TGenericObject>(
```

Comment:
Looks like entity must be of type CORE:File.

can add a check in this plugin and throw error if `entity["type"] !== "dmss://system/SIMOS/Blueprint"`